fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java

```diff
@@ -26,16 +29,50 @@
  *
  */
 public abstract class ContentObject extends RepositoryObject {
+	
+	// the object containing MODS and (possibly also) source metadata
+	private FileObject fileObj;
+
+	private BinaryObject mods;
 
 	protected ContentObject(PID pid, Repository repository, RepositoryObjectDataLoader dataLoader) {
 		super(pid, repository, dataLoader);
 	}
 
-	public BinaryObject addDescription(InputStream contentStream) {
-		return null;
+	public FileObject addDescription(InputStream modsStream) {
+		fileObj = createFileObject();
+		
+		BinaryObject mods = fileObj.addOriginalFile(modsStream, null, "text/xml", null);
+		repository.createRelationship(pid, IanaRelation.describedby, mods.getResource());
+		
+		return fileObj;
+	}
+	
+	public FileObject addDescription(InputStream sourceMdStream, String sourceProfile,
+			InputStream modsStream) {
+		fileObj = createFileObject();
+		
+		BinaryObject orig = fileObj.addOriginalFile(sourceMdStream, null, "text/plain", null);
```

Comment:
In this case, the original file is a source metadata file, so in addition to its normal types, it should also be assigned Cdr.SourceMetadata as an RDF.type